policy.go

```diff
@@ -173,7 +174,11 @@ func (p *defaultPolicy) Add(key uint64, cost int64) ([]*item, bool) {
 		}
 
 		// If the incoming item isn't worth keeping in the policy, reject.
-		if incHits < minHits {
+		// However, a little bit of randomness is introduced to prevent against
+		// hash collision attacks, in which the score of the victims is artificially
+		// raised so that no new items are admitted into the cache.
+		// This strategy is implemented by Caffeine.
+		if incHits < minHits && !(rand.Int() % 127 == 0) {
```

Comment:
You can replace the modulus `%` with a bitmask `&`. This was why 127 was chosen (2^n - 1) so that we could avoid the division at ~1% acceptance rate. The division vs mask was perhaps an over-optimization in this code path, though!